torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -1563,6 +1567,60 @@ ComputeAtMap::getAllDisjointSetConsumers(
   return visited;
 }
 
+void IterDomainGraph::updateComputeWith(TensorView* compute_with_tv) {
+  TORCH_INTERNAL_ASSERT(
+      compute_with_tv->hasResolvedComputeWith(),
+      "Invalid tensor: ",
+      compute_with_tv->toString());
+
+  // Can use any consumer this tensor is computed with
```

Comment:
I thought compute with is resolved after compute at map is built but this reads to me like we need to resolve the compute with before building the compute at map. I'm a little confused on the logical dependencies.

Reply:
Okay, think I understood now:

1. Compute at map is built
2. Expressions are sorted
3. Compute with (the producer) is resolved based on first consumer found in expression sorting
4. Compute at map updates the loop mappings based on compute with
5. Expressions are sorted again with the compute with information
The only step that concerns me I think is step (5). It seems to me we should assert the expressions are sorted the same way as in (2) otherwise I think that would be a logical inconsistency with the approach.

Loop map being updated in (4) I believe is to generate the right loop structure not to modify the sorted expression order.

Reply:
Didn't specify, @naoyam this is my only "sticking point" before approving.

Reply:
Yes, that is correct. I'm not sure how the second sort could result in inconsistency. As long as the sorting work as intended, I believe we should be fine. The second sorting could result in different ordering as some more IterDomains are loop-mapped with updated max producer positions, but as long as it completes the soring algorithm, isn't the sorted list valid?

Reply:
Yeah, I'm okay with this. The expectation is the second sorting wouldn't have a tangible difference, some things I guess could be reordered, but I would think that shouldn't actually be the case because loop map would be updated based on the  first consumer after a producer and updating the loop map should only reinforce them being next to each other.

I'm just wondering if we should assert the first and second sorting match, but maybe given the algorithm has some degrees of freedom it might just trigger coincidentally some expr "group" to reorder with another expr "group" which is just that "degree of freedom". i.e. there may be more than one unique valid ordering.

Yeah, I'll change this to approve, I'm okay with this.

Reply:
Yes, there's some freedom, and updated loop mapping and max producer position might affect the decision about which ordering to use within the freedom. Otherwise, everything should be deterministic (except for bugs of course). I'm reasonably confident that the expression sorting should be able to handle both the first and second sorting consistently.